docs/blog/2021-01-19-frontend-asset-passthrough.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Frontend Asset Passthrough
+authors:
+  - name: Mike Cutalo
+    url: https://github.com/mcutalo88
+    avatar: https://avatars1.githubusercontent.com/u/2250844?s=460&u=24deb32096e9f892cc91a6ff1ca1af50193b1fbd&v=4
+description: Frontend Asset Passthrough
+image: https://user-images.githubusercontent.com/4712430/104760766-7a2c5980-5727-11eb-93f5-3296b23ba3a0.png
+hide_table_of_contents: false
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Ever wonder how Clutch handles serving frontend assets without a CDN in a distributed multi-version deployment?
+This article will touch on how Lyft deploys Clutch and some of the early problems we faced when rolling it out.
+
+<!--truncate-->
+
+## Problem
+
+When deploying Clutch at Lyft we noticed very early on when deploying a new version,
+a subset of users would fail to load the webpage as some of the frontend assets could not be resolved.
+
+Clutch bundles all frontend assets into a single binary, which allows us to serve all frontend assets from disk, simplifying our architecture by omitting the need for a CDN.
+However, this means that during deploys, particularly when deploying to canary (which at Lyft is a particular subset of the production environment),
+there will be an intermediary state in which the frontend assets requested by the user may not exist on the hosts serving that request while the rollout is still progressing, since during deploys, there will briefly exist two different versions of the application taking traffic.
+
+Typically, organizations solve this problem by simply adding a CDN to their service architecture and serve static assets from there.
+However, we decided to go a different route, as the Clutch architecture and design philosophy values simplicity and avoiding adding new dependencies unless absolutely necessary. This simplicity is core to Clutch's values as we want to keep Clutch easy to deploy and configure and cloud provider agnostic, with as few external dependencies as possible.
```

Comment:
thought it was a lil repetitive to mention simplicity again but could just be me
```suggestion
However, we decided to go a different route, as the Clutch architecture and design philosophy values simplicity and avoiding adding new dependencies unless absolutely necessary. We want to keep Clutch easy to deploy,  configure, and cloud provider agnostic, with as few external dependencies as possible.
```

Reply:
still reads a bit weird, "We want to keep Clutch easy to ... cloud provider agnostic"

Reply:
Yeah I think this is also supposed to be `We want to keep Clutch easy to deploy and configure, and cloud provider agnostic, ...`

Reply:
Added in some of the suggestions lmk.